cmd/telegraf/telegraf.go

```diff
@@ -100,7 +105,21 @@ Examples:
   telegraf -config telegraf.conf -input-filter cpu:mem -output-filter influxdb
 `
 
-func main() {
+var logger service.Logger
+
+var stop chan struct{}
+
+var srvc service.Service
+
+type program struct{}
+
+func reloadLoop(stop chan struct{}, s service.Service) {
+	defer func() {
```

Comment:
`service.Interactive()` is a boolean function which detects if the program is being run interactively (as opposed to as a service). The purpose of the `defer` function is to cause the program to exit when the main function does if it is being run interactively. It was added in response to a bug where the program would never exit if you simply asked it to provide information (ie `telegraf -help`).